shotover/src/transforms/kafka/sink_cluster/node.rs

```diff
@@ -141,10 +141,15 @@ impl ConnectionFactory {
             ));
         }
 
+        let delegation_token = scram_over_mtls
+            .token_task
+            .get_token_for_user(scram_over_mtls.username.clone())
+            .await?;
+
         // SCRAM client-first
         let mut scram = Scram::<Sha256>::new(
-            scram_over_mtls.delegation_token.token_id.clone(),
-            scram_over_mtls.delegation_token.hmac.to_string(),
+            delegation_token.token_id.clone(),
```

Comment:
```suggestion
            delegation_token.token_id,
```
We can avoid the clone here since we own `delegation_token` and dont use it again after this line, the `token_id` will just be moved out of `delegation_token`.

Reply:
Ah yes, I'll remove the clone